locallib.php

```diff
@@ -47,7 +47,11 @@ function subcourse_available_courses($userid = null) {
     }
 
     $fields = 'fullname,shortname,idnumber,category,visible,sortorder';
-    $mycourses = get_user_capability_course('moodle/grade:viewall', $userid, true, $fields, 'sortorder');
+
+    $mycourses = \core_course_category::search_courses(
+        ['limittoenrolled' => get_config('mod_subcourse', 'limittoenrolled')],
+        ['moodle/grade:viewall']
+    );
 
```

Comment:
I am a bit confused by this. I tried to look at the given method `\core_course_category::search_courses()` but its signature suggests that it is not accepting parameters such as `limittoenrolled`. A parameter of that name is used in the external function, but not in this method. Unless I am missing something...

Reply:
I see, I messed both of them up here - I'll have a look at it again.
But also changing to `\core_course_category::search_courses()` alone improves performance a lot (factor 12 here for first call, second call is cached).

We have an instance here with > 400000 courses where using the limittoenrolled function (of the external function) is the only way to keep course searching functional.